deeppavlov/models/go_bot/tracker/featurized_tracker.py

```diff
@@ -105,3 +122,152 @@ def _new_features(self, state) -> np.ndarray:
+            given the path to stories.md naively read steps from it. ToDo use MDYAML reader
+            Args:
+                story_path: the path to stories.md
+
+            Returns:
+                the dict containing info on all the stories used
+            """
+            story_f = open(story_path, 'r')
+            stories_li = []
+            curr_story = None
+            for line in story_f:
+                line = line.strip()
+                if not line: continue;
+                if line.startswith("#"):
+                    if curr_story is not None:
+                        stories_li.append(curr_story)
+                    story_name = line.strip('#').strip()
+                    curr_story = {"story": story_name, "steps": []}
+                elif line.startswith("*"):
+                    # user turn
+                    step = {"intent": line.strip('*').strip()}
+                    curr_story["steps"].append(step)
+                elif line.startswith('-'):
+                    # system turn
+                    step = {"action": line.strip('-').strip()}
+                    curr_story["steps"].append(step)
+            if curr_story is not None:
+                stories_li.append(curr_story)
+            story_f.close()
+            stories_di = {"stories": stories_li}
+            return stories_di
+
+        stories_md_path = expand_path(stories_yml_path)
+        stories_yml_di = read_md_story(stories_md_path)
+        prev_forms = []
+        action2forms = {}
+        for story in stories_yml_di["stories"]:
+            story_name = story["story"]
+            story_steps = story["steps"]
+            for step in story_steps:
+                if "action" not in step.keys():
+                    continue
+
+                curr_action = step["action"]
+                if curr_action.startswith("form"):
+                    curr_action = json.loads(curr_action[len("form"):])["name"]
+                    print(curr_action)
+                if curr_action in form_names:
+                    prev_forms.append(curr_action)
+                if curr_action in potential_api_or_db_actions:
+                    action2forms[curr_action] = prev_forms
+                    prev_forms = []
+
+        def get_slots(system_utter: str, form_name: str) -> List[str]:
+            """
+            Given the utterance story line, extract slots information from it
+            Args:
+                system_utter: the utterance story line
+                form_name: the form we are filling
+
+            Returns:
+                the slots extracted from the line
+            """
+            slots = []
+            if system_utter.startswith(f"utter_ask_{form_name}_"):
+                slots.append(system_utter[len(f"utter_ask_{form_name}_"):])
+            elif system_utter.startswith(f"utter_ask_"):
+                slots.append(system_utter[len(f"utter_ask_"):])
+            else:
+                # todo: raise an exception
```

Comment:
What prevents you from raise exception now?

Reply:
I don't have time to properly handle it...